common/client.go

```diff
@@ -46,19 +46,20 @@ type RestClientConfig struct {
 	Retries       int
 	Credential    *Credential
 	TestMode      bool
+	RootURL       string
 }
 
-func GetDefaultRestClientConfig() RestClientConfig {
-	return RestClientConfig{TimeoutMillis: DefaultRestTimeout, Retries: DefaultRestRetries}
+func GetDefaultRestClientConfig(rootURL string) RestClientConfig {
+	return RestClientConfig{TimeoutMillis: DefaultRestTimeout, Retries: DefaultRestRetries, RootURL: rootURL}
 }
 
 // GetRestClientConfig returns a RestClientConfig based on a ServiceConfig
 func GetRestClientConfig(config ServiceConfig) RestClientConfig {
-	return RestClientConfig{TimeoutMillis: config.Common.Api.RestTimeoutMillis, Retries: config.Common.Api.RestRetries}
+	return RestClientConfig{TimeoutMillis: config.Common.Api.RestTimeoutMillis, Retries: config.Common.Api.RestRetries, RootURL: config.Common.Api.RootServiceUrl}
 }
 
 // NewRestClient creates a new Rest client.
```

Comment:
These docstrings could be made more useful. It's kind of clear from the name that this function creates a new REST client. That's what the function name says. But what does this entail? What is this new REST client configured with? What is it used for? Only for our services? Can it be used for any arbitrary REST service? What assumptions does it make? If that's documented somewhere else, at least refer to that here.

We are looking for information that's useful to a reader.